### ISSUE TITLE
Removing the sklearn dependency

### ISSUE THREAD

**Issue by jvmncs** [OpenMined/PySyft#2535]
Currently, there's a scikit-learn dependency for PySyft, however it's only being used in two places -- the `create_sandbox` feature in sandbox.py, which loads a few datasets from sklearn, and a tutorial on federated learning where it's being used as a shortcut to calculating ROC.  This seems like a heavy dependency for accomplishing such simple and standard tasks. Removing it would improve build times and lessen the code surface area for e.g. enterprise audits.

In order to remove the dependency, we need to fix the two code snippets that mention it:
- The federated learning tutorial can have a note suggesting that the user install sklearn to follow along.
- The sandbox is only slightly more involved. I suggest we add an exception catch for ImportError around any sklearn imports -- if the ImportError is caught, we print a friendly reminder to install sklearn in order to use the sandbox.

**Comment by kakirastern** [comment]
I would like to take on/be assigned this issue.

**Comment by jvmancuso** [comment]
Hi, sorry! I have actually tackled this issue in an unsubmited PR for
installing PySyft with different frameworks (torch, TFE keras, and
tensorflow so far). I'm still waiting to polish it before submitting, but
should be in early this week.

**Comment by kakirastern** [comment]
No worries @jvmancuso

**Comment by Ankit-Dhankhar** [comment]
@jvmancuso  What do you think about completely removing sklearn dependency (instead of throwing Import Error) and in-place of that use torchvision like module for loading datasets?

If this sounds good I can make required changes for that?

**Comment by jvmancuso** [comment]
Hm -- on the one hand, I like the idea of moving to torchvision because the datasets are a bit less trivially solved than the ones in sklearn.  On the other hand, we're moving away from having pytorch be a required dependency (see for example [this RFC](https://github.com/OpenMined/rfcs/blob/master/20190808-modular-pysyft-frameworks.md) and [these PRs](https://github.com/OpenMined/PySyft/pulls?utf8=%E2%9C%93&q=generalize)).  Similarly, one might want to create a TF version of the sandbox in the PySyft-TensorFlow repo once that code is ready.

I think we should keep the sklearn sandbox here for now, since people seem to use it.  But I also think once we move the frameworks out to PySyft-PyTorch and PySyft-TensorFlow, those repos might want to have their own versions of the sandbox that are a bit more suitable for experimenting with deep learning.

What do you think?

**Comment by ribhu97** [comment]
Hi @jvmancuso Is this issue still open? I would like to be assigned to this if no one is currently working on it.

**Comment by agarwalmanvi** [comment]
Is this still open?

**Comment by jvmncs** [comment]
Hey all, this was finished as a result of https://github.com/OpenMined/PySyft/pull/2636

Going to close this, feel free to open up a new issue if there are any further concerns!
